testing/test_pytester.py

```diff
@@ -419,66 +422,64 @@ def test_preserve_container(self, monkeypatch, path_type) -> None:
         assert getattr(sys, other_path_type) == original_other_data
 
 
-def test_testdir_subprocess(testdir) -> None:
-    testfile = testdir.makepyfile("def test_one(): pass")
-    assert testdir.runpytest_subprocess(testfile).ret == 0
+def test_pytester_subprocess(pytester: Pytester) -> None:
+    testfile = pytester.makepyfile("def test_one(): pass")
+    assert pytester.runpytest_subprocess(testfile).ret == 0
 
 
-def test_testdir_subprocess_via_runpytest_arg(testdir) -> None:
-    testfile = testdir.makepyfile(
+def test_pytester_subprocess_via_runpytest_arg(pytester: Pytester) -> None:
+    testfile = pytester.makepyfile(
         """
-        def test_testdir_subprocess(testdir):
+        def test_pytester_subprocess(pytester):
             import os
-            testfile = testdir.makepyfile(
+            testfile = pytester.makepyfile(
                 \"""
                 import os
                 def test_one():
                     assert {} != os.getpid()
                 \""".format(os.getpid())
             )
-            assert testdir.runpytest(testfile).ret == 0
+            assert pytester.runpytest(testfile).ret == 0
         """
     )
-    result = testdir.runpytest_subprocess(
-        "-p", "pytester", "--runpytest", "subprocess", testfile
-    )
+    result = pytester.runpytest("-p", "pytester", "--runpytest", "subprocess", testfile)
```

Comment:
I think this should be run via `pytester.runpytest()` rather then `pytester.runpytest_subprocess()` to utilize `"--runpytest", "subprocess"` args? 

Added coverage suggests the same -> [src/_pytest/pytester.py#L1145](https://codecov.io/gh/pytest-dev/pytest/src/9bdd7512db9844af81ac5c8601f8a58e3e92086d/src/_pytest/pytester.py#L1145) (Codecov link)

Reply:
Right, I think it makes sense.

Actually, `pytester.runpytest` translates to either `runpytest_subprocess` or `runpytest_inline` depending on the `--runpytest` argument (which is what is being tested here...). So might even want to use `runpytest_inline` directly here.

The coverage thing I think is just because we don't collect coverage from subprocesses. There's an issue for this #7679.